crates/apub/src/activities/mod.rs

```diff
@@ -190,45 +186,41 @@ where
   Url::parse(&id)
 }
 
+pub(crate) trait GetActorType {
+  fn actor_type(&self) -> ActorType;
+}
+
 #[tracing::instrument(skip_all)]
 async fn send_lemmy_activity<Activity, ActorT>(
   data: &Data<LemmyContext>,
   activity: Activity,
   actor: &ActorT,
-  mut inbox: Vec<Url>,
+  send_targets: ActivitySendTargets,
   sensitive: bool,
 ) -> Result<(), LemmyError>
 where
   Activity: ActivityHandler + Serialize + Send + Sync + Clone,
-  ActorT: Actor,
+  ActorT: Actor + GetActorType,
   Activity: ActivityHandler<Error = LemmyError>,
 {
-  static CACHE: Lazy<Cache<(), Arc<Vec<String>>>> = Lazy::new(|| {
-    Cache::builder()
-      .max_capacity(1)
-      .time_to_live(DEAD_INSTANCE_LIST_CACHE_DURATION)
-      .build()
-  });
-  let dead_instances = CACHE
-    .try_get_with((), async {
-      Ok::<_, diesel::result::Error>(Arc::new(Instance::dead_instances(&mut data.pool()).await?))
-    })
-    .await?;
-
-  inbox.retain(|i| {
-    let domain = i.domain().expect("has domain").to_string();
-    !dead_instances.contains(&domain)
-  });
```

Comment:
If this dead instance check isnt used anymore, the code for `update_instance_software()` in scheduled tasks can be simplified.

Reply:
It is still used, the usage is just now moved to crates/federate/lib.rs. Dead instances are now fully excluded from having federation workers started for them instead of their inboxes being filtered out.

But: It is a good point, since the structure has changed, I can remove the Instance::get_dead call.